src/corelibs/U2Algorithm/src/misc/FindAlgorithm.cpp

```diff
@@ -155,6 +156,10 @@ static void findInAmino(FindAlgorithmResultsListener* rl,
     QByteArray revPattern(pattern);
     TextUtils::reverse(revPattern.data(), patternLen);
 
+    if (!DynTable::isAcceptableMatrixDimensions(width, height)) {
+        coreLog.error(QObject::tr("Matrix with given dimensions too big for calculation."));
```

Comment:
User does not know what is the 'matrix'. Change the text: 'The search pattern is too long.'